src/_P088_HeatpumpIR.ino

```diff
@@ -43,47 +43,9 @@
  *
  */
 
-#include <AUXHeatpumpIR.h>
-#include <BalluHeatpumpIR.h>
-#include <CarrierHeatpumpIR.h>
-#include <DaikinHeatpumpIR.h>
-#include <FuegoHeatpumpIR.h>
-#include <FujitsuHeatpumpIR.h>
-#include <GreeHeatpumpIR.h>
-#include <HisenseHeatpumpIR.h>
-#include <HitachiHeatpumpIR.h>
-#include <HyundaiHeatpumpIR.h>
-#include <IVTHeatpumpIR.h>
-#include <MideaHeatpumpIR.h>
-#include <MitsubishiHeatpumpIR.h>
-#include <MitsubishiHeavyFDTCHeatpumpIR.h>
-#include <MitsubishiHeavyHeatpumpIR.h>
-#include <MitsubishiMSCHeatpumpIR.h>
-#include <MitsubishiSEZKDXXHeatpumpIR.h>
-#include <PanasonicCKPHeatpumpIR.h>
-#include <PanasonicHeatpumpIR.h>
-#include <SamsungHeatpumpIR.h>
-#include <SharpHeatpumpIR.h>
-#include <ToshibaDaiseikaiHeatpumpIR.h>
-#include <ToshibaHeatpumpIR.h>
-
-// Array with all supported heatpumps
-HeatpumpIR *heatpumpIR[] = {new PanasonicCKPHeatpumpIR(), new PanasonicDKEHeatpumpIR(), new PanasonicJKEHeatpumpIR(),
-                            new PanasonicNKEHeatpumpIR(), new PanasonicLKEHeatpumpIR(),
-                            new CarrierNQVHeatpumpIR(), new CarrierMCAHeatpumpIR(),
-                            new MideaHeatpumpIR(), new FujitsuHeatpumpIR(),
-                            new MitsubishiFDHeatpumpIR(), new MitsubishiFEHeatpumpIR(), new MitsubishiMSYHeatpumpIR(), new MitsubishiFAHeatpumpIR(),
-                            new MitsubishiKJHeatpumpIR(), new MitsubishiHeavyFDTCHeatpumpIR(),
-                            new SamsungAQVHeatpumpIR(), new SamsungFJMHeatpumpIR(),new SharpHeatpumpIR(), new DaikinHeatpumpIR(),
-                            new MitsubishiHeavyZJHeatpumpIR(), new MitsubishiHeavyZMHeatpumpIR(),
-                            new MitsubishiSEZKDXXHeatpumpIR(), new MitsubishiMSCHeatpumpIR(),
-                            new HyundaiHeatpumpIR(), new HisenseHeatpumpIR(),
-                            new GreeGenericHeatpumpIR(), new GreeYANHeatpumpIR(), new GreeYAAHeatpumpIR(),
-                            new FuegoHeatpumpIR(), new ToshibaHeatpumpIR(), new ToshibaDaiseikaiHeatpumpIR(),
-                            new IVTHeatpumpIR(), new HitachiHeatpumpIR(),
-                            new BalluHeatpumpIR(), new AUXHeatpumpIR(),
-                            NULL};
+#include <HeatpumpIRFactory.h>
 
+HeatpumpIR *heatpumpIR;
```

Comment:
You're now using the `heatpumpIR` pointer only at a single spot, so you can consider only having the pointer declared locally where you call the factory.
Maybe even consider making it something like a smart pointer, so it will call the delete when it gets out of scope, making sure you never forget to call the `delete`.

Reply:
Moved into the local scope and initialized to NULL.